### DISCUSSION TITLE
Is it possible to add marked.js extensions to  some plugin for parcel?

### DISCUSSION THREAD

**Discussion by hrbigelow** [parcel-bundler/parcel#9254] [General]
Hi,

I would like to use one of the parcel-transformer-markdown or similar plugin to parse something like:

```js
import md from './content.md'
div.innerHtml = md
```

Unfortunately, I don't think that parcel-transformer-markdown supports adding custom extensions to marked.js via their extension mechanism:

[marked use](https://marked.js.org/using_pro)

I tried writing one, but I can't figure out how to actually get the parsed content.

```js
import {marked} from 'marked';
import markedMathExt from './marked_math_ext.js';
import {Transformer} from '@parcel/plugin';

var mm = new markedMathExt(true)
marked.use({ extensions: mm.getExtensions() });



var transformer = new Transformer({
  async transform({asset}) {
    console.log(`In plugin transform function: Got asset\n${asset}`)
    // Retrieve the asset's source code and source map.
    let code = await asset.getCode();
    // let sourceMap = await asset.getMap();
    let options = {
      breaks: true,
      pedantic: false,
      gfm: true,
      tables: true,
      headerIds: false,
      mangle: false,
      sanitize: false,
      smartLists: true,
      smartypants: false,
      xhtml: false,
    }

    // Run it through some compiler, and set the results on the asset.
    code = marked.parse(code, options);
    asset.type = 'html';
    asset.setCode(code);
    // asset.setMap(map);

    // Return the asset
    return [asset];
  }
});
```

When I tried using this, the import variable:

`import md from './content.md'`

is just a URI of a bundled resource.  Is there some way I can get the actual HTML content output by the plugin?

Thanks very much!

**Comment by mischnic** [comment]
Does `import md from 'bundle-text:./content.md'` work?
Docs: https://parceljs.org/features/bundle-inlining/#inlining-a-bundle-as-text
